opm/io/eclipse/OutputStream.hpp

```diff
@@ -357,6 +359,75 @@ namespace Opm { namespace EclIO { namespace OutputStream {
                        const std::vector<T>& data);
     };
 
+    class SummarySpecification
+    {
+    public:
+        using StartTime = std::chrono::system_clock::time_point;
+
+        enum class UnitConvention
+        {
+            Metric = 1,
+            Field  = 2,
+            Lab    = 3,
+            Pvt_M  = 4,
+        };
+
+        struct RestartSpecification
+        {
+            std::string root;
+            int step;
+        };
+
+        class Parameters
+        {
+        public:
+            void add(const std::string& keyword,
+                     const std::string& name,
+                     const int          num,
+                     const std::string& unit);
+
+            friend class SummarySpecification;
+
+        private:
+            std::vector<PaddedOutputString<8>> keywords{};
+            std::vector<PaddedOutputString<8>> names{};
+            std::vector<int>                   nums{};
+            std::vector<PaddedOutputString<8>> units{};
+        };
+
+        explicit SummarySpecification(const ResultSet&            rset,
+                                      const Formatted&            fmt,
+                                      const UnitConvention        uconv,
+                                      const std::array<int,3>&    cartDims,
+                                      const RestartSpecification& restart,
+                                      const StartTime             start);
+
+        ~SummarySpecification();
+
+        SummarySpecification(const SummarySpecification& rhs) = delete;
+        SummarySpecification(SummarySpecification&& rhs);
+
+        SummarySpecification& operator=(const SummarySpecification& rhs) = delete;
+        SummarySpecification& operator=(SummarySpecification&& rhs);
+
+        void write(const Parameters& params);
+
+    private:
+        int unit_;
+        int restartStep_;
+        std::array<int,3> cartDims_;
+        std::vector<int> startDate_;
```

Comment:
`std::chrono<something>`?

Reply:
> `std::chrono<something>`?

Maybe[tm].  It's a trade-off between memory requirements and calculating the 6-element broken down start date required for SMSPEC's `STARTDAT` array on each call to `write()`.